docs/request-custom.md

```diff
@@ -0,0 +1,210 @@
+# Implement a custom request
+
+This lesson describes how to implement your own custom request types, for types that
+don't have out-of-the-box Volley support.
+
+## Write a custom request
+
+Most requests have ready-to-use implementations in the toolbox; if your response is a string,
+image, or JSON, you probably won't need to implement a custom `Request`.
+
+For cases where you do need to implement a custom request, this is all you need
+to do:
```

Comment:
comment that applies to the whole PR: is there a consistent line length limit we're formatting these docs to? It seems pretty inconsistent here - for example, moving this line to the previous line still wouldn't be as long as the previous line.

I can dig into markdown formatter options once this is merged, though, so we can try to handle this automatically.

Reply:
I kept the original line lengths wherever possible (which was apparently 85-ish), but I didn't auto-format this markdown. Can be tidied up later.